pytest_splunk_addon/splunk.py

```diff
@@ -485,45 +487,123 @@ def sc4s(request):
     """
     if request.config.getoption("splunk_type") == "external":
         request.fixturenames.append("uf_external")
         uf = request.getfixturevalue("uf_external")
-    elif request.config.getoption("splunk_type") == "docker":
-        request.fixturenames.append("uf_docker")
-        uf = request.getfixturevalue("uf_docker")
+    elif request.config.getoption("splunk_type") == "kubernetes":
+        request.fixturenames.append("uf_kubernetes")
+        uf = request.getfixturevalue("uf_kubernetes")
     else:
         raise Exception
     yield uf
 
 
 @pytest.fixture(scope="session")
-def uf_docker(docker_services, tmp_path_factory, worker_id, request):
+def uf_kubernetes(request):
     """
     Provides IP of the uf server and management port based on pytest-args(splunk_type)
     """
-    LOGGER.info("Starting docker_service=uf")
-    os.environ["CURRENT_DIR"] = os.getcwd()
-    if worker_id:
-        # get the temp directory shared by all workers
-        root_tmp_dir = tmp_path_factory.getbasetemp().parent
-        fn = root_tmp_dir / "pytest_docker"
-        with FileLock(str(fn) + ".lock"):
-            docker_services.start("uf")
+    if (
+        "PYTEST_XDIST_WORKER" not in os.environ
+        or os.environ.get("PYTEST_XDIST_WORKER") == "gw0"
+    ):
+        LOGGER.info(
+            "Starting kubernetes_service=uf for worker id {}".format(
+                str(os.environ.get("PYTEST_XDIST_WORKER"))
+            )
+        )
+        current_path = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            "k8s_manifests",
+            "uf",
+        )
+        os.environ["ADDON_LOCATION"] = str(
+            os.path.join(
+                os.getenv("TEST_RUNNER_DIRECTORY"),
+                "tests",
+                "src",
+                str(os.getenv("SPLUNK_ADDON")),
+            )
+        )
+        if ":\\" in os.getenv("ADDON_LOCATION"):
+            os.environ["ADDON_LOCATION"] = os.getenv("ADDON_LOCATION").split(":\\", 1)[
+                1
+            ]
+        if "\\" in os.getenv("ADDON_LOCATION"):
+            os.environ["ADDON_LOCATION"] = os.getenv("ADDON_LOCATION").replace(
+                "\\", "/"
+            )
+        if ":\\" in os.getenv("TEST_RUNNER_DIRECTORY"):
+            os.environ["TEST_RUNNER_DIRECTORY_UF"] = os.getenv(
+                "TEST_RUNNER_DIRECTORY"
+            ).split(":\\", 1)[1]
+        else:
+            os.environ["TEST_RUNNER_DIRECTORY_UF"] = os.getenv("TEST_RUNNER_DIRECTORY")
+        if "\\" in os.getenv("TEST_RUNNER_DIRECTORY_UF"):
+            os.environ["TEST_RUNNER_DIRECTORY_UF"] = os.getenv(
+                "TEST_RUNNER_DIRECTORY_UF"
+            ).replace("\\", "/")
+        update_k8s_manifest_files(folder=current_path, file="uf_deployment")
+        LOGGER.info("Setting up UF")
+        kubernetes_helper_uf = KubernetesHelper()
+        kubernetes_helper_uf.create_k8s_resource_from_yaml(
```

Comment:
Is it possible to create resources in parallel to make it faster?

Reply:
SC4S depends on Splunk setup, I think it won't be possible for now.